src/js/letters/containers/AddressSection.jsx

```diff
@@ -223,6 +242,7 @@ export class AddressSection extends React.Component {
         <div>
           <Address
             onInput={this.handleChange}
+            onBlur={this.dirtyInput}
```

Comment:
Probably totally fine, but worth pointing out that both `this.dirtyInput` and `this.handleChange` call `setState`, with the results of the `setState` call in`onBlur` getting reused in the `setState` call in `handleChange`. Since these are async I suppose we could get into a weird state where the two are out of sync, but I'm not sure that's an actual problem - everything seems to work as expected in local testing and hopefully there's enough time between `onBlur` triggering and `onInput` triggering that this'll never come up.

Reply:
Yeah, I was trying to figure out how to abstract that in a way that made sense, but I came to that same conclusion--they're not going to get messed up with normal use. The only way I can think of for the state to get weird is if `dirtyInput` is called before `handleChange` is finished, but that would take _really_ fast input and blur. Even then, the out-of-sync-ness (there's probably a proper word for that...) isn't a big deal, I don't think, since the next edit will ensure everything's back to normal.

I'm pretty sure, anyhow.